sql/importer: Block IMPORT INTO on RLS tables

spilchen opened the pull request:
Tables enabled for row-level security (RLS) have policies that restrict which rows can be added. Since IMPORT bypasses the query engine where these policies are enforced, we need to block IMPORT INTO for RLS tables. There is one exception: admin users can still run this command, as RLS policies are not enforced for administrators.

Epic: CRDB-45203
Release note: none
Informs #136807

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/141844)
<!-- Reviewable:end -->

spilchen committed:
sql/importer: Block IMPORT INTO on RLS tables

Tables enabled for row-level security (RLS) have policies that restrict
which rows can be added. Since IMPORT bypasses the query engine where
these policies are enforced, we need to block IMPORT INTO for RLS
tables. There is one exception: admin users can still run this command,
as RLS policies are not enforced for administrators.

Epic: CRDB-45203
Release note: none
Informs #136807

Dedej-Bergin reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/141844#-:-OJsqlTU97CSIy0Npp6Y:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->
